envoy/upstream/upstream.h

```diff
@@ -811,6 +811,11 @@ class ClusterInfo : public Http::FilterChainFactory {
    */
   virtual const absl::optional<std::chrono::milliseconds> idleTimeout() const PURE;
 
+  /**
```

Comment:
how does tihs differ from idleTimeout?  At a glance it looks to me like it's possible all we need is a tcp options parallel to the http protocol options so we can set idleTimeout for non-tcp-upstreams. WDIT?

Reply:
I assumed you were saying `non-http-upstreams`

`idleTimeout` here is the same idea with `tcpPoolIdleTimeout` so we can use `idleTimeout` for both http codec and tcp pool.

`idleTimeout` could be what http protocol options say or what cluster proto says, as we need separated configs. Does it make sense?

Reply:
I eventually keep it separated since i'm not sure if we can have a cluster has both idleTimeout for http and tcpPoolIdleTimeout for tcp pool. happy to make it to one if it's not possible to have both